test_refreshable_mv/test.py::test_backup_inner_table is flaky

```
test_refreshable_mv/test.py:443: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

to_table = False

    def do_test_backup(to_table):
        for node in nodes:
            node.query(
                "create database re engine = Replicated('/test/re', 'shard1', '{replica}');"
            )
    
        target = 'rmv'
        if to_table:
            node1.query("create table re.tgt (x Int64) engine ReplicatedMergeTree order by x")
            target = 'tgt'
    
        node1.query(
            "create table re.src (x Int64) engine ReplicatedMergeTree order by x;"
            "insert into re.src values (1);")
        node2.query(
            "system sync database replica re;"
            "system sync replica re.src;"
            f"create materialized view re.rmv refresh every 1 second {'TO re.tgt' if to_table else '(x Int64) engine ReplicatedMergeTree order by x'} as select x from re.src;"
            "system wait view re.rmv")
        assert node2.query(f"select * from re.{target}") == "1\n"
    
        backup_destination = new_backup_destination()
        tables_exist_query = "SELECT count() FROM system.tables where database='re' AND name in ('src', 'rmv')"
    
        node1.query(
            f"BACKUP ALL EXCEPT DATABASE system ON CLUSTER 'default' TO {backup_destination};"
        )
        for node in nodes:
            node.query(
                "drop database re sync;"
                "drop database if exists re_broken_replicated_tables sync;")
        assert node1.query(tables_exist_query) == "0\n"
    
        node1.query(f"RESTORE ALL ON CLUSTER 'default' FROM {backup_destination};")
    
        assert node1.query(tables_exist_query) == "2\n"
        assert node2.query(tables_exist_query) == "2\n"
        node1.query(f'SYSTEM SYNC REPLICA re.{target}')
        node2.query(f'SYSTEM SYNC REPLICA re.{target}')
>       assert node1.query(f'SELECT * FROM re.{target}') == '1\n'
E       AssertionError: assert '' == '1\n'
E         - 1
```

Cc @al13n321 
CI: https://s3.amazonaws.com/clickhouse-test-reports/json.html?PR=80719&sha=c5c0bc5f545194878202871acf54d242cfc958c1&name_0=PR&name_1=Integration%20tests%20%28aarch64%2C%20distributed%20plan%2C%204%2F4%29

--- a/src/Backups/BackupUtils.cpp
+++ b/src/Backups/BackupUtils.cpp
@@ -126,8 +126,8 @@ bool isInnerTable(const QualifiedTableName & table_name)
 
 bool isInnerTable(const String & /* database_name */, const String & table_name)
 {
-    /// We skip inner tables of materialized views.
-    return table_name.starts_with(".inner.") || table_name.starts_with(".inner_id.");
+    /// We skip inner tables of materialized views. They're backed up by StorageMaterializedView.
+    return table_name.starts_with(".inner.") || table_name.starts_with(".inner_id.") || table_name.starts_with(".tmp.inner.") || table_name.starts_with(".tmp.inner_id.");
 }
 
 }

--- a/tests/integration/test_refreshable_mv/test.py
+++ b/tests/integration/test_refreshable_mv/test.py
@@ -424,8 +424,13 @@ def do_test_backup(to_table):
 
     assert node1.query(tables_exist_query) == "2\n"
     assert node2.query(tables_exist_query) == "2\n"
-    node1.query(f'SYSTEM SYNC REPLICA re.{target}')
-    node2.query(f'SYSTEM SYNC REPLICA re.{target}')
+    if not to_table:
+        # Inner tables are not backed up, wait for first refresh.
+        node1.query(f'SYSTEM WAIT VIEW re.{target}')
+        node2.query(f'SYSTEM WAIT VIEW re.{target}')
+    else:
+        node1.query(f'SYSTEM SYNC REPLICA re.{target}')
+        node2.query(f'SYSTEM SYNC REPLICA re.{target}')
     assert node1.query(f'SELECT * FROM re.{target}') == '1\n'
     assert node2.query(f'SELECT * FROM re.{target}') == '1\n'
 

